packages/cli/src/zosfiles/compare/CompareBaseHelper.ts

```diff
@@ -0,0 +1,230 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { IHandlerParameters, DiffUtils } from "@zowe/imperative";
+import {  IZosFilesResponse } from "@zowe/zos-files-for-zowe-sdk";
+
+/**
+ * Base helper for compare command to handle common operations through the comparison
+ * @export
+ */
+export default class ComapareBaseHelper {
+
+    /**
+     * Singleton instance of this class
+     * @private
+     * @static
+     * @type {ComapareBaseHelper}
+     * @memberof ComapareBaseHelper
+     */
+    private static hInstance: ComapareBaseHelper;
+
+    /**
+     * Return a singleton instance of this class
+     * @static
+     * @readonly
+     */
+    public static get instance(): ComapareBaseHelper {
+        if(this.hInstance == null){
+            this.hInstance = new ComapareBaseHelper();
+        }
+
+        return this.hInstance;
+    }
```

Comment:
In the future, Zowe CLI daemon mode may support executing multiple commands in parallel. If multiple handlers running at the same time share a singleton class, this could cause race conditions.

Instead of using a singleton class, I think it would be better to construct a new instance of the class each time it is used in a handler (`new CompareBaseHelper()`).